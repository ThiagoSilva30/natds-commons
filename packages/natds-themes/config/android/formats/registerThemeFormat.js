import path from 'path';
import { allPass, filter } from 'ramda';
import {
  flattenProps, isAssetFile, isAsset, negate, formatBuilder,
} from '../../shared/helpers';
import {
  isDimension, createEncodedHashFromValue, isColor, mapFilteredValues,
} from '../shared/helpers';

const encodeValue = (item) => ({ ...item, value: createEncodedHashFromValue(item.value) });

const templateDataBuilder = (
  { properties }, platform, { brandName, mode },
) => {
  const props = flattenProps(properties);

  const filterPropsWithoutPrefix = allPass([
    negate(isColor),
    negate(isDimension),
    negate(isAsset),
  ]);

  return {
    brandName,
    colors: mapFilteredValues(props, isColor, encodeValue),
    dimensions: mapFilteredValues(props, isDimension, encodeValue),
    drawables: filter(isAssetFile),
    materialMode: mode === 'Light' ? 'Light' : 'DayNight',
    mode,
    properties: filter(filterPropsWithoutPrefix),
  };
};

const registerThemeFormat = () => {
  const templatePath = path.resolve(__dirname, './templates/themes.hbs');

  return formatBuilder('android/themes', templatePath, templateDataBuilder);
};

export default registerThemeFormat;
