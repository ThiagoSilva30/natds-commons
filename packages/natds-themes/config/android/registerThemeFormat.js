import path from 'path';
import { formatBuilder } from '../shared/formatBuilder';
import { flattenProps } from '../shared/helpers';
import {
  isDimension, createEncodedHashFromValue, isColor, mapFilteredValues,
} from './helpers';

const encodeValue = (item) => ({ ...item, value: createEncodedHashFromValue(item.value) });

const templateDataBuilder = (
  { properties }, platform, { brandName, mode },
) => {
  const props = flattenProps(properties);

  const otherProps = props.filter((item) => !isColor && !isDimension(item));

  return {
    brandName,
    colors: mapFilteredValues(props, isColor, encodeValue),
    dimensions: mapFilteredValues(props, isDimension, encodeValue),
    materialMode: mode === 'Light' ? 'Light' : 'DayNight',
    mode,
    properties: otherProps,
  };
};

const registerThemeFormat = () => {
  const templatePath = path.resolve(__dirname, './templates/themes.hbs');

  return formatBuilder('android/themes', templatePath, templateDataBuilder);
};

export default registerThemeFormat;
