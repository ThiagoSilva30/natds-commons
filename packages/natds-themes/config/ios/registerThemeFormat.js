import path from 'path';
import { formatBuilder } from '../shared/formatBuilder';
import { flattenProps, splitTokensAndComponents } from '../shared/helpers';

const registerThemeFormat = () => {
  const templatePath = path.resolve(__dirname, './templates/theme.hbs');

  const templateDataBuilder = ({ properties }) => {
    const { tokens, components } = splitTokensAndComponents(properties);

    return {
      components: flattenProps(components),
      tokens: flattenProps(tokens),
    };
  };

  return formatBuilder('ios-swift/theme.swift', templatePath, templateDataBuilder);
};

export default registerThemeFormat;
