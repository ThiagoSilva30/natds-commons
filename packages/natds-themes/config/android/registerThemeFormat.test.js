import registerThemeFormat from './registerThemeFormat';

const dictionary = {
  properties: {
    color: {
      primary: {
        attributes: {},
        name: 'colorPrimary',
        original: {
          value: '#F091C9',
        },
        path: [
          'color',
          'primary',
        ],
        value: '#F091C9',
      },
    },
  },
};

const platform = {
  files: [
    {
      destination: 'theme_attributes.xml',
    },
    {
      destination: 'Theme.Avon.Dark.xml',
    },
  ],
};

describe('registerAttributesFormat', () => {
  it('should return the register config', () => {
    const register = registerThemeFormat();

    expect(typeof register.formatter).toBe('function');
    expect(register.name).toBe('android/themes');
  });

  it('should return the correct formatter template', () => {
    const register = registerThemeFormat();

    expect(register.formatter(dictionary, platform)).toMatchSnapshot();
  });
});
