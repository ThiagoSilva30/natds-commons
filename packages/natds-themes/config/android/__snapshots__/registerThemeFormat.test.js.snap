// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registerAttributesFormat should return the correct formatter template 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<resources>
  <style name=\\"Theme.Avon.Dark\\" parent=\\"Theme.MaterialComponents.DayNight\\">
    <item name=\\"colorPrimary\\">#F091C9</item>
  </style>
</resources>"
`;
